data-plane/core/src/main/java/dev/knative/eventing/kafka/broker/core/utils/BaseEnv.java

```diff
@@ -4,15 +4,26 @@
 
 import java.util.function.Function;
 
-public abstract class BaseEnv {
+public class BaseEnv {
 
   public static final String PRODUCER_CONFIG_FILE_PATH = "PRODUCER_CONFIG_FILE_PATH";
   public static final String DATA_PLANE_CONFIG_FILE_PATH = "DATA_PLANE_CONFIG_FILE_PATH";
 
+  public static final String METRICS_PORT = "METRICS_PORT";
+  public static final String METRICS_PATH = "METRICS_PATH";
+  public static final String METRICS_PUBLISH_QUANTILES = "METRICS_PUBLISH_QUANTILES";
+
   private final String producerConfigFilePath;
   private final String dataPlaneConfigFilePath;
 
+  private final int metricsPort;
+  private final String metricsPath;
+  private final boolean metricsPublishQuantiles;
+
   public BaseEnv(Function<String, String> envProvider) {
+    this.metricsPath = requireNonNull(envProvider.apply(METRICS_PATH));
```

Comment:
Can we have the metrics configuration optional? If you don't add the envs for that, the metrics options are just not created and the metrics listener doesn't start. WDYT?

Reply:
It makes sense, however, I would like to have a more concise way of disabling metrics through a ConfigMap regardless of the env variable's values. 
The main reason for this is that to disable metrics users need to change every deployment, `kafka-broker-receiver`, `kafka-broker-dispatcher`, and `kafka-sink-receiver` (and more are coming), and that doesn't seem optimal to me. 

The `ConfigMap` format can be:
```yaml
metrics.enabled: (true|false)
```
or 
```properties
metrics.enabled=(true|false)
```

WDYT? Should we add a `ConfigMap`?

Reply:
Do we have some "monitoring" config map in eventing that we can use to check if metrics are enabled or not?

Reply:
No, `config-observability` hasn't any `enable`/`disable` flag for metrics:

```yaml
    # metrics.backend-destination field specifies the system metrics destination.
    # It supports either prometheus (the default) or stackdriver.
    # Note: Using stackdriver will incur additional charges
    metrics.backend-destination: prometheus

    # metrics.request-metrics-backend-destination specifies the request metrics
    # destination. If non-empty, it enables queue proxy to send request metrics.
    # Currently supported values: prometheus, stackdriver.
    metrics.request-metrics-backend-destination: prometheus

    # metrics.stackdriver-project-id field specifies the stackdriver project ID. This
    # field is optional. When running on GCE, application default credentials will be
    # used if this field is not provided.
    metrics.stackdriver-project-id: "<your stackdriver project id>"

    # metrics.allow-stackdriver-custom-metrics indicates whether it is allowed to send metrics to
    # Stackdriver using "global" resource type and custom metric type if the
    # metrics are not supported by "knative_broker", "knative_trigger", and "knative_source" resource types.
    # Setting this flag to "true" could cause extra Stackdriver charge.
    # If metrics.backend-destination is not Stackdriver, this is ignored.
    metrics.allow-stackdriver-custom-metrics: "false"

    # profiling.enable indicates whether it is allowed to retrieve runtime profiling data from
    # the pods via an HTTP server in the format expected by the pprof visualization tool. When
    # enabled, the Knative Eventing pods expose the profiling data on an alternate HTTP port 8008.
    # The HTTP context root for profiling is then /debug/pprof/.
    profiling.enable: "false"

    # sink-event-error-reporting.enable whether the adapter reports a kube event to the CRD indicating
    # a failure to send a cloud event to the sink.
    sink-event-error-reporting.enable: "false"
```

We can propose to add it, it requires a bit of work to implement it tho.

Reply:
well, let me ask another question then, is it possible in eventing to disable metrics collection? If not, then we don't allow disabling them too and that's it

Reply:
I think it's not possible ATM.